File: server/bcephal-loader/src/main/java/com/moriset/bcephal/loader/domain/FileLoaderEditorData.java
/**
 * 
 */
package com.moriset.bcephal.loader.domain;

import java.util.ArrayList;
import java.util.List;

import com.moriset.bcephal.domain.BrowserData;
import com.moriset.bcephal.domain.EditorData;
import com.moriset.bcephal.domain.Nameable;

import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * @author Joseph Wambo
 *
 */
@Data
@EqualsAndHashCode(callSuper = false)
public class FileLoaderEditorData extends EditorData<FileLoader> {

	private List<BrowserData> grids;
	
	private List<BrowserData> materializedGrids;

	private List<BrowserData> tables;
	
	private List<Nameable> routines = new ArrayList<Nameable>();
	
	private List<String> dateFormats;
	
	private List<Nameable> templates;

}

File: client/bcephal-client-model/Loaders/FileLoaderEditorData.cs
using Bcephal.Models.Base;
using Bcephal.Models.Grids;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bcephal.Models.Loaders
{
    public class FileLoaderEditorData : EditorData<FileLoader>
    {

        public ObservableCollection<BrowserData> Grids { get; set; }

        public ObservableCollection<BrowserData> Tables { get; set; }

        public ObservableCollection<Nameable> Routines { get; set; }

    }
}
